src/server/detail/snapshot_storage.cc

```diff
@@ -32,9 +33,19 @@ namespace detail {
 using namespace util;
 using namespace std;
 
-// Returns bucket_name, obj_path for an s3 path.
-optional<pair<string, string>> GetBucketPath(string_view path) {
-  std::string_view clean = absl::StripPrefix(path, kS3Prefix);
+inline bool IsGcsPath(string_view path) {
+  return absl::StartsWith(path, kGCSPrefix);
+}
+
+constexpr string_view kSummarySuffix = "summary.dfs"sv;
+
+pair<string, string> GetBucketPath(string_view path) {
+  string_view clean = path;
+  if (absl::StartsWith(clean, kS3Prefix)) {
+    clean = absl::StripPrefix(clean, kS3Prefix);
+  } else {
+    clean = absl::StripPrefix(clean, kGCSPrefix);
+  }
```

Comment:
I'd prefer to write it in a more generic way 
const auto& prefix = absl::StartsWith(clean, kS3Prefix) ? kS3Prefix : kGCSPrefix;
clean = absl::StripPrefix(clean, prefix).

Reply:
👍🏼  will do that in the next PR.